test/wilson/dom_test.cljs

```diff
@@ -177,6 +178,20 @@
                 [:td "false"]
                 [:td "true"]]]]))))
 
+(deftest sorted-table-test
+  (let [rows [{:a 1 :b "B" :c -6}
+               {:a 4 :b "A" :c 2}
+               {:a 3 :b "D" :c 4}]
+        ks (d/prepare-keys [:a :b :c])
+        component (d/sorted-table ks rows)
+        [table-headers table-rows] (match component
+                                     [:table _
+                                      [:thead
+                                       [:tr & table-headers]]
+                                      [:tbody & table-rows]]
+                                     [table-headers table-rows])]
```

Comment:
@smogg This was where the bug was. Apparently core.match got very confused because rows was being used as a binding variable in a match clause, and it was bound in the same surrounding `let`. I need to do more experiments to reproduce a minimal case.